### PR TITLE
Alerting: Default to right instant or range when filling alert rule from query params

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes the alert rule form not updating correctly the instant or range field in the query when pre-filling from query params.
In case of be ommited, we should default to the right values.

**Why do we need this feature?**

This pre-filed fields should match the query params.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

related to this [escalation](https://github.com/grafana/support-escalations/issues/12764) 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by gillesdemey — APPROVED** [review]
code changes LGTM :rocket:

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx [hunk 1]
```diff
@@ -401,6 +404,30 @@ function formValuesFromPrefill(rule: Partial<RuleFormValues>): RuleFormValues {
...
 }
 
+function setInstantOrRange(values: RuleFormValues): RuleFormValues {
+  return {
+    ...values,
+    queries: values.queries?.map((query) => {
+      if (!isExpressionQuery(query.model)) {
+        // data query
+        const defaultToInstant =
+          query.model.datasource?.type === 'loki' || query.model.datasource?.type === 'prometheus';
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx:414]: Can we use `DataSourceType.Loki` and `DataSourceType.Prometheus`?

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx [hunk 2]
```diff
@@ -401,6 +404,30 @@ function formValuesFromPrefill(rule: Partial<RuleFormValues>): RuleFormValues {
   });
 }
 
+function setInstantOrRange(values: RuleFormValues): RuleFormValues {
+  return {
+    ...values,
+    queries: values.queries?.map((query) => {
+      if (!isExpressionQuery(query.model)) {
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx:411]: Also we could early return in the opposite case perhaps?

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx (hunk 1 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx:414]: 💯 I was not aware of this enum

**Review by tomratcliffe — APPROVED** [review]
🚀

**Pull request merged**